### ISSUE TITLE
Discuss possible refactor

### ISSUE THREAD

**Issue by johntdyer** [william20111/go-thousandeyes#32]
Hey William,

 I wanted to open the door to us having a discussion around the project structure and see if you would entertain a refactor of the code base to maybe make it more manageable ? 

My thought process is is that we structure the project by domains, kinda like the AWS SDK does... what are your thoughts on this ? 

-John

**Comment by william20111** [comment]
Yes, orginal was to support test creation for the terraform provider. If we look to add more functionality and try to support everything the API offers makes sense to have better structure for that. +1

**Comment by johntdyer** [comment]
Truthfully I am most interested in the TF provider myself but I find that implementing it involves several facts of the API that are not necessarily "tests", for example agents, monitors, alerts, ect...   I noticed you're in Scotland, was there a good time we could sync up on webex  ?

**Comment by william20111** [comment]
Yes, happy to grab some time. Sunday would work for me so whatever time works best can carve out a bit time to have conversation via webex.
